databind/src/main/java/gov/nist/secauto/metaschema/databind/io/json/MetaschemaJsonWriter.java

```diff
@@ -297,8 +302,14 @@ private <T extends IFeatureComplexItemValueHandler & IBoundInstanceModel> void w
 
       IBoundInstanceFlag jsonKey = handler.getItemJsonKey(parentItem);
       if (jsonKey != null) {
+        Object keyValue = jsonKey.getValue(parentItem);
+        if (keyValue == null) {
+          throw new IOException(String.format("Null JSON key value for definition '%s'",
```

Comment:
Minor, non-blocking: given the above similar feedback maybe?

```suggestion
          throw new IOException(String.format("Null value for json-key for definition '%s'",
```